packages/filesystem/src/node/node-filesystem.ts

```diff
@@ -306,7 +307,24 @@ export class FileSystemNode implements FileSystem {
         if (moveToTrash) {
             return trash([FileUri.fsPath(_uri)]);
         } else {
-            return fs.remove(FileUri.fsPath(_uri));
+            const filePath = FileUri.fsPath(_uri);
+            const outputRootPath = paths.join(os.tmpdir(), v4());
+            try {
+                await new Promise<FileStat>((resolve, reject) => {
+                    fs.rename(filePath, outputRootPath, async error => {
+                        if (error) {
+                            return reject(error);
+                        }
+                        resolve(await this.doGetStat(FileUri.create(outputRootPath), 1));
```

Comment:
btw why do you need to fetch `doGetStat` here?

Reply:
> btw why do you need to fetch `doGetStat` here? you don't use it afterwards

You are correct.  I have removed it.